vendor/k8s.io/utils/clock/testing/fake_clock.go

```diff
@@ -0,0 +1,294 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/utils/clock"
+)
+
+var (
+	_ = clock.PassiveClock(&FakePassiveClock{})
+	_ = clock.Clock(&FakeClock{})
+	_ = clock.Clock(&IntervalClock{})
+)
+
+// FakePassiveClock implements PassiveClock, but returns an arbitrary time.
+type FakePassiveClock struct {
```

Comment:
This whole file was directly copied from:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apimachinery/pkg/util/clock/clock.gohttps://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apimachinery/pkg/util/clock/clock.go

Please don't do that. We don't want to maintain two copies of the same code.

If there are methods that you need additionally (I see you added Tick method) - please add it there. And get rid of this file completely.

Reply:
This comment is about a vendored file, not a change in this PR.
The work done in k8s.io/utils/clock was to make that at least as good as the apimachinery clock package, so that we can retire the latter.  See #94738 , https://github.com/kubernetes/utils/pull/213 , and https://github.com/kubernetes/utils/pull/197 .

Reply:
Oops sorry - I missed it's in vendor/.